src/components/enterprise-subsidy-requests/data/hooks.js

```diff
@@ -49,41 +50,95 @@ export function useSubsidyRequests(subsidyRequestConfiguration) {
   const [couponCodeRequests, setCouponCodeRequests] = useState([]);
   const [isLoading, setIsLoading] = useState(false);
 
-  useEffect(() => {
-    const fetchSubsidyRequests = async (subsidyType) => {
-      setIsLoading(true);
-      try {
-        if (subsidyType === SUBSIDY_TYPE.COUPON) {
-          const { data: { results } } = await fetchCouponCodeRequests(
-            subsidyRequestConfiguration.enterpriseCustomerUuid,
-          );
-          const requests = camelCaseObject(results);
-          setCouponCodeRequests(requests);
-        } if (subsidyType === SUBSIDY_TYPE.LICENSE) {
-          const { data: { results } } = await fetchLicenseRequests(
-            subsidyRequestConfiguration.enterpriseCustomerUuid,
-          );
-          const requests = camelCaseObject(results);
-          setLicenseRequests(requests);
-        }
-      } catch (error) {
-        logError(error);
-      } finally {
-        setIsLoading(false);
+  const fetchSubsidyRequests = async (subsidyType) => {
+    setIsLoading(true);
+    try {
+      if (subsidyType === SUBSIDY_TYPE.COUPON) {
+        const { data: { results } } = await fetchCouponCodeRequests(
+          subsidyRequestConfiguration.enterpriseCustomerUuid,
+        );
+        const requests = camelCaseObject(results);
+        setCouponCodeRequests(requests);
+      } if (subsidyType === SUBSIDY_TYPE.LICENSE) {
+        const { data: { results } } = await fetchLicenseRequests(
+          subsidyRequestConfiguration.enterpriseCustomerUuid,
+        );
+        const requests = camelCaseObject(results);
+        setLicenseRequests(requests);
       }
-    };
+    } catch (error) {
+      logError(error);
+    } finally {
+      setIsLoading(false);
+    }
+  };
 
+  const loadSubsidyRequests = () => {
     if (subsidyRequestConfiguration?.subsidyRequestsEnabled) {
       const { subsidyType } = subsidyRequestConfiguration;
       if (subsidyType) {
         fetchSubsidyRequests(subsidyType);
       }
     }
+  };
+
+  useEffect(() => {
+    loadSubsidyRequests();
```

Comment:
Is there a chance that `subsidyRequestConfiguration?.subsidyRequestsEnabled` changes from `true` to `false`, and thus kicks off this `useEffect` to load more subsidy requests, even though subsidy requests are now disabled?

Does this need a `if (subsidyRequestConfiguration?.subsidyRequestsEnabled) { ... }` check?

Reply:
Yes, we check that in the `loadSubsidyRequests` function.

Reply:
D'oh, thanks for clarifying :)